play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -257,7 +267,15 @@ private[jackson] class PlaySerializers(parserSettings: JsonParserSettings) exten
 }
 
 private[json] object JacksonJson {
-  private lazy val mapper = (new ObjectMapper).registerModule(new PlayJsonModule(JsonParserSettings.settings))
+  /**
+   * Instance used to serialize and deserialize JSON. This is configured with system properties, but can be
+   * overridden for testing.
+   */
+  var instance: JacksonJson = JacksonJson(JsonParserSettings.settings)
```

Comment:
Global static state isn't ideal, but this isn't globally exposed and adds a hook for testing.

I think it would be better to use implicit state, but I couldn't figure out a way to do that without breaking backwards compatibility.

Reply:
Used `AtomicReference`

Reply:
Reverted the `AtomicReference` change, as mentioned in another thread.

All serialization or parsing operations will be done on a single immutable `JacksonJson` object, and any changes here will not affect any such operations in flight. Concurrent tests that change the config could produce unexpected results, but the only way to avoid that is to refactor the library to avoid static state. I don't think that's a problem in practice, though.